File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/infrastructure/entity/CreatorProfileEntity.java
package com.eager.questioncloud.core.domain.creator.infrastructure.entity;

import com.eager.questioncloud.core.domain.creator.model.CreatorProfile;
import com.eager.questioncloud.core.domain.question.enums.Subject;
import jakarta.persistence.Embeddable;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Embeddable
@AllArgsConstructor
@NoArgsConstructor
public class CreatorProfileEntity {
    @Enumerated(EnumType.STRING)
    private Subject mainSubject;
    private String introduction;

    public static CreatorProfileEntity from(CreatorProfile creatorProfile) {
        return new CreatorProfileEntity(creatorProfile.getMainSubject(), creatorProfile.getIntroduction());
    }

    public CreatorProfile toModel() {
        return CreatorProfile.create(mainSubject, introduction);
    }
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/infrastructure/entity/CreatorProfileEntity.kt
package com.eager.questioncloud.core.domain.creator.infrastructure.entity

import com.eager.questioncloud.core.domain.creator.model.CreatorProfile
import com.eager.questioncloud.core.domain.question.enums.Subject
import jakarta.persistence.Embeddable
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated

@Embeddable
class CreatorProfileEntity private constructor(
    @Enumerated(EnumType.STRING)
    var mainSubject: Subject,
    var introduction: String
) {
    fun toModel(): CreatorProfile {
        return CreatorProfile.create(mainSubject, introduction)
    }

    companion object {
        @JvmStatic
        fun from(creatorProfile: CreatorProfile): CreatorProfileEntity {
            return CreatorProfileEntity(creatorProfile.mainSubject, creatorProfile.introduction)
        }
    }
}
